NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -0,0 +1,41 @@
+abstract type AbstractCategory end
+
+abstract type AbstractGroup <: AbstractCategory end
+
+label(c::AbstractCategory) = error("method `label` not defined for type $(typeof(c))")
+
+function dimension(c::AbstractCategory)
+  return error("method `dimension` not defined for category or group $(typeof(c))")
+end
+
+function fusion_rule(::Type{C}, l1, l2) where {C<:AbstractCategory}
+  return error("fusion_rule not defined for type $(C)")
+end
+
+function fusion_rule(c1::AbstractCategory, c2::AbstractCategory)
+  C = typeof(c1)
+  return [C(d) for d in fusion_rule(C, label(c1), label(c2))]
+end
+
+⊗(c1::AbstractCategory, c2::AbstractCategory) = fusion_rule(c1, c2)
+
+Base.:(*)(s1::AbstractCategory, s2::AbstractCategory) = ⊗(s1, s2)
+
+⊕(a::AbstractCategory, b::AbstractCategory) = [a, b]
+⊕(v::Vector{<:AbstractCategory}, b::AbstractCategory) = vcat(v, b)
+⊕(a::AbstractCategory, v::Vector{<:AbstractCategory}) = vcat(a, v)
+
+function Base.show(io::IO, q::Vector{<:AbstractCategory})
+  isempty(q) && return nothing
+  symbol = ""
+  for l in q
+    print(io, symbol, l)
+    symbol = " ⊕ "
+  end
+end
+
+function trivial(::Type{C}) where {C<:AbstractCategory}
```

Comment:
Can this just be defined on instances, i.e. `trivial(::AbstractCategory)`?

Reply:
Never mind, I think this makes sense, I see you forward `istrivial(::AbstractCategory)` to this one so either one can be used.

Reply:
However, I have preferred the style:
```julia
function trivial(categorytype::Type{<:AbstractCategory})
  return error("method `trivial` not defined for type $categorytype")
end
```

Reply:
I think that's a better style too, and originally had it, but I put back the placeholder name `C` just so I could print it out as part of the error message.